docs/contribution_guide.md

```diff
@@ -77,3 +77,20 @@ https://captum.ai/tutorials/House_Prices_Regression_Interpret
 **Multimodal**
 - You can use VQA model and dataset described here:
 https://captum.ai/tutorials/Multimodal_VQA_Captum_Insights
+
+
+## Docstring style
+
+Docstring is required for all public APIs to provide users the details of the arguments and returns. [Our API documentation](https://captum.ai/api/) is generated from the docstring. Captum adopts a customized docstring format modified on top of [Google style](https://www.sphinx-doc.org/en/master/usage/extensions/example_google.html). Specifically, each argument should be listed as `arg_name (type): description` in the `Args:` section. The argument typing convention:
+- primitive types: `int`, `str`, `float`, `bool`
+- common collection types: `list`, `tuple`, `dict`
+    - [PEP 585](https://peps.python.org/pep-0585/#implementation) has deprecated the duplicate types: `List`, `Tuple`, `Dict`
+    - element types: `list[int]`, `dict[int, str]`
+- other foundamental types: `Any`, `Callable`, `Interable`
+- class types: `MyClass`, `external_lib.SomeClass`
+- omit `torch` for common Pytorch types: `Tensor`, `nn.Module`
```

Comment:
Unless Tensor and nn are imported in the Python file, this is going to break the Sphinx hyperlinking.

If we want to display `Tensor`, then we should be using `~torch.Tensor`. The `~torch.` could be appended while generating the docs, or be present by default in docstrings themselves.

Reply:
For `nn.Module`, I think the Sphinx compatible options are `torch.nn.Module`, or  `~torch.nn.Module` (displayed as simply `Module`).

Reply:
The pr is just to document the agreed styles that the team prefers to write docstring within the codebase. 

For how sphinx compiles it or how it will be displayed in the webpage, we can leave them to the future. I don't think sphinx has any issues to handle even free text typing. So for "compatible", I guess you only mean hyperlinks, which are not that important anyway given our webpage is not hyperlinked much (as well as other torch libs). We can figure out it later.

Reply:
@aobo-y Yeah, for this issue I already have the code for turning `Tensor` into `~torch.Tensor` during HTML creation, so it that's basically a solved problem.

The usage of `nn.Module` however doesn't work as there is no way to ensure the intersphinx extension always recognizes it correctly, and manual replacement strategies cannot reproduce the right styling. It may show up as hyperlinked in files where `import torch.nn as nn` is used, which will then be inconsistent across Captum as not every file does that.

I would say that it's easiest to just include `torch.` with `nn.Module` for now, and then it can be changed in the future if a better solution is found. Future formatting changes would then decrease line lengths, meaning it'll be easier to do without fixing docstring line length linting errors.

Reply:
great then! At least, we are sure about `Tensor`.

Inconsistent is ok for now. Just like no file is currently following the rules here. We can progressively fix them next. Rules here are for how developer write docstring within the code. The compiled page may look different. For example, "nn.Module" can be rendered as "torch.nn.Module". We will figure out when we move to improve the building process.